src/web/server/app/services/auth.ts

```diff
@@ -153,3 +160,78 @@ export const approveRegistration = (
 export const removePendingAccount = (username: string) => {
   client.execute(DELETE_PENDINGACCOUNTS_ALL, [username], { prepare: true });
 };
+
+/**
+ * Store the token and userId into password_reset_tokens table
+ * @params token - The password reset token
+ * @params userId - The user ID associated with the token
+ */
+export const storeResetToken = async (token: string, userId: string) => {
+  await client.execute(INSERT_PASSWORDRESETTOKENS, [token, userId], {
+    prepare: true,
+  });
+};
+
+/**
+ * Retrieve the userId and email using username as lookup
+ * @params username - The username of the user
+ * @returns ResultSet - Contains the user_id and email field of the user
+ */
+export const retrieveUserIdAndEmail = async (username: string) => {
+  return client.execute(SELECT_USERSNAME_USERID_EMAIL, [username], {
+    prepare: true,
+  });
+};
+
+/**
+ * Verify the reset token exists in the table
+ * @params token - The token to be verified
+ * @returns boolean - True if token is valid, false otherwise
+ */
+export const verifyToken = async (token: string) => {
+  const result = await client.execute(SELECT_PASSWORDRESETTOKENS, [token], {
+    prepare: true,
+  });
+  return result.rows.length === 1;
+};
+
+/**
+ * Retrieve the associated user ID using the reset token
+ * @params token - The password reset token
+ * @returns any - The id of the retrieved user
```

Comment:
Isn't the type a string?

Reply:
I thought so too, but when I hover my cursor over the return value, it said `any`, so I wanted to be consistent with what it said.

Reply:
@melvinheks So should I put string or any?

Reply:
string probably